src/plugins/intel_gpu/src/kernel_selector/cl_kernels/slice_ref.cl

```diff
@@ -4,33 +4,89 @@
+
 #if INPUT0_DIMS <= 4
-    const uint xy = get_global_id(2);
-    const uint y = xy / OUTPUT_SIZE_X;
-    const uint x = xy % OUTPUT_SIZE_X;
-    const uint output_index = OUTPUT_GET_INDEX(batch, feature, y, x);
-    const uint input_index = INPUT0_GET_INDEX(
-        SLICE_BEGIN_BATCH + batch * SLICE_STEP_BATCH,
-        SLICE_BEGIN_FEATURE + feature * SLICE_STEP_FEATURE,
-        SLICE_BEGIN_Y + y * SLICE_STEP_Y,
-        SLICE_BEGIN_X + x * SLICE_STEP_X);
+    const long slice_begin_dim2 = BRING_INTO_RANGE(slice_start[2], INPUT0_SIZE_Y);
+    const long slice_begin_dim3 = BRING_INTO_RANGE(slice_start[3], INPUT0_SIZE_X);
+    const long output_dim23 = get_global_id(2);
+    const long output_dim2 = output_dim23 / OUTPUT_SIZE_X;
+    const long output_dim3 = output_dim23 % OUTPUT_SIZE_X;
+    const long output_index = OUTPUT_GET_INDEX(output_dim0, output_dim1, output_dim2, output_dim3);
+    const long input_index = INPUT0_GET_INDEX(
+        slice_begin_dim0 + output_dim0 * slice_step[0],
+        slice_begin_dim1 + output_dim1 * slice_step[1],
+        slice_begin_dim2 + output_dim2 * slice_step[2],
+        slice_begin_dim3 + output_dim3 * slice_step[3]);
 #elif INPUT0_DIMS == 5
-    const uint xyz = get_global_id(2);
-    const uint yx = xyz % (OUTPUT_SIZE_X * OUTPUT_SIZE_Y);
-    const uint z = xyz / (OUTPUT_SIZE_X * OUTPUT_SIZE_Y);
-    const uint y = yx / OUTPUT_SIZE_X;
-    const uint x = yx % OUTPUT_SIZE_X;
-    const uint output_index = OUTPUT_GET_INDEX(batch, feature, z, y, x);
-    const uint input_index = INPUT0_GET_INDEX(
-        SLICE_BEGIN_BATCH + batch * SLICE_STEP_BATCH,
-        SLICE_BEGIN_FEATURE + feature * SLICE_STEP_FEATURE,
-        SLICE_BEGIN_Z + z * SLICE_STEP_Z,
-        SLICE_BEGIN_Y + y * SLICE_STEP_Y,
-        SLICE_BEGIN_X + x * SLICE_STEP_X);
+    const long slice_begin_dim2 = BRING_INTO_RANGE(slice_start[2], INPUT0_SIZE_Z);
+    const long slice_begin_dim3 = BRING_INTO_RANGE(slice_start[3], INPUT0_SIZE_Y);
+    const long slice_begin_dim4 = BRING_INTO_RANGE(slice_start[4], INPUT0_SIZE_X);
+    const long output_dim234 = get_global_id(2);
+    const long output_dim34 = output_dim234 % (OUTPUT_SIZE_X * OUTPUT_SIZE_Y);
+    const long output_dim2 = output_dim234 / (OUTPUT_SIZE_X * OUTPUT_SIZE_Y);
+    const long output_dim3 = output_dim34 / OUTPUT_SIZE_X;
+    const long output_dim4 = output_dim34 % OUTPUT_SIZE_X;
+    const long output_index = OUTPUT_GET_INDEX(output_dim0, output_dim1, output_dim2, output_dim3, output_dim4);
+    const long input_index = INPUT0_GET_INDEX(
+        slice_begin_dim0 + output_dim0 * slice_step[0],
+        slice_begin_dim1 + output_dim1 * slice_step[1],
+        slice_begin_dim2 + output_dim2 * slice_step[2],
+        slice_begin_dim3 + output_dim3 * slice_step[3],
+        slice_begin_dim4 + output_dim4 * slice_step[4]);
 #endif
+
     output[output_index] = ACTIVATION(input[input_index], ACTIVATION_PARAMS);
 }
+
+#undef LOAD_BUFFER;
+#undef BRING_INTO_RANGE;
```

Comment:
nit: please add new line at the end of the file

Reply:
Fixed.